receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper.go

```diff
@@ -0,0 +1,154 @@
+	"github.com/open-telemetry/opentelemetry-collector/receiver/hostmetricsreceiver/internal"
+)
+
+// Scraper for CPU Metrics
+type Scraper struct {
+	config    *Config
+	consumer  consumer.MetricsConsumer
+	startTime pdata.TimestampUnixNano
+}
+
+// NewCPUScraper creates a set of CPU related metrics
+func NewCPUScraper(ctx context.Context, cfg *Config, consumer consumer.MetricsConsumer) (*Scraper, error) {
+	return &Scraper{config: cfg, consumer: consumer}, nil
+}
+
+// Start
+func (c *Scraper) Start(ctx context.Context) error {
+	bootTime, err := host.BootTime()
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		c.startTime = pdata.TimestampUnixNano(bootTime)
+
+		ticker := time.NewTicker(c.config.CollectionInterval())
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				c.scrapeMetrics(ctx)
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	return nil
+}
+
+// Close
+func (c *Scraper) Close(ctx context.Context) error {
+	return nil
+}
+
+func (c *Scraper) scrapeMetrics(ctx context.Context) {
+	ctx, span := trace.StartSpan(ctx, "cpuscraper.scrapeMetrics")
+	defer span.End()
+
+	metricData := data.NewMetricData()
+	metrics := internal.InitializeMetricSlice(metricData)
+
+	err := c.scrapeAndAppendMetrics(metrics)
+	if err != nil {
+		span.SetStatus(trace.Status{Code: trace.StatusCodeDataLoss, Message: fmt.Sprintf("Error(s) when scraping cpu metrics: %v", err)})
+		return
+	}
+
+	if metrics.Len() > 0 {
+		err := c.consumer.ConsumeMetrics(ctx, pdatautil.MetricsFromInternalMetrics(metricData))
+		if err != nil {
+			span.SetStatus(trace.Status{Code: trace.StatusCodeDataLoss, Message: fmt.Sprintf("Unable to process metrics: %v", err)})
+			return
+		}
+	}
+}
+
+func (c *Scraper) scrapeAndAppendMetrics(metrics pdata.MetricSlice) error {
+	cpuTimesStats, err := cpu.Times(c.config.ReportPerCPU)
+	if err != nil {
+		return err
+	}
+
+	cpuSecondsData := flattenCPUTimesStatData(cpuTimesStats)
+	metric := internal.AddNewMetric(metrics)
+	initializeCPUSecondsMetric(metric, c.startTime, cpuSecondsData)
+	return nil
+}
+
+type cpuSecondsData struct {
+	cpuLabel   string
+	stateLabel string
+	value      int64
+}
+
+func flattenCPUTimesStatData(cpuTimes []cpu.TimesStat) []cpuSecondsData {
+	result := []cpuSecondsData{}
+
+	for _, cpuTime := range cpuTimes {
+		result = append(result, cpuSecondsData{cpuLabel: cpuTime.CPU, stateLabel: UserStateLabelValue, value: int64(cpuTime.User)})
+		result = append(result, cpuSecondsData{cpuLabel: cpuTime.CPU, stateLabel: SystemStateLabelValue, value: int64(cpuTime.System)})
+		result = append(result, cpuSecondsData{cpuLabel: cpuTime.CPU, stateLabel: IdleStateLabelValue, value: int64(cpuTime.Idle)})
+		result = append(result, cpuSecondsData{cpuLabel: cpuTime.CPU, stateLabel: InterruptStateLabelValue, value: int64(cpuTime.Irq)})
+		result = append(result, cpuSecondsData{cpuLabel: cpuTime.CPU, stateLabel: IowaitStateLabelValue, value: int64(cpuTime.Iowait)})
+	}
+
+	return result
+}
```

Comment:
This is unnecessary (to construct the result slice), you can directly add points.